docs/versioning.rst

```diff
@@ -0,0 +1,55 @@
+*****************
+Versioning Policy
+*****************
+
+The internal interface of the `ddtrace` library is defined as follows:
+    Any module, function, class or attribute is internal if it is prefixed with a single underscore
+
+    Any package that is named `internal` is internal
+
+    Any package, module, function, class or attribute that is contained within a namespace that is internal is also internal
+
+When a new major version is released it becomes the primary release line, where all new features, bug fixes and security patches land.
+
+The non-negative integers components of the *vMAJOR.MINOR.PATCH* format are incremented by the following rules:
+
+MAJOR
+    API changes incompatible with previous versions
+
+    Functionality changes incompatible with previous versions
+
+    Dropping support for anything, but not limited to, Python versions, library versions supported in integrations, or Datadog APM features.
```

Comment:
> library versions supported in integrations

🤔 I am not sure we figured this part out yet.

Keeping in `MINOR` is good, but maybe we should remove from here.

e.g. if we dropped support for an older version of Django we probably wouldn't do a major version bump (we _might_ want to, but we haven't really defined our process there yet, I'd hate to commit to it without exploring further)

Reply:
Agree -- I put it here but with that same question.

Reply:
Can we still commit to dropping support for a Python version only in major versions?

Reply:
100% that is definitely a breaking change that requires a major version bump.